addons/mail/models/ir_http.py

```diff
@@ -29,3 +31,18 @@ def session_info(self):
                 'user_context': user_context,
             })
         return result
+
+    def _document_check_access(self, model_name, document_id, access_token=None):
```

Comment:
This is apparently moved from portal, shouldn't it also be *re*moved from there?

Reply:
I believe I did remove it from portal, at least the code itself, not the method.

However I didn't adapt all existing occurrences but simply did a forward for now: https://github.com/odoo/odoo/pull/82912/files#diff-2acd0e6f67a1190985466e669bc815626745d14981784d828e404f8855af186eR366

Do you suggest that I go over the 26 existing occurrences across various modules to make the use the new method instead and get rid of the old one completely? That looked a bit overkill in stable, but if you think it's better I can do it.

Reply:
Not sure what 'model_name' is, but as anyway you expect an access_token field and it seems called only on ir attachment, why not moving it as a model method on attachment model ? Not sure why we need this kind of generic methods on ir_http, nothing to do with routing code unless I misunderstood something ?

Reply:
This is a move of the portal method, it's called with all kind of `model_name` as parameter in the various portal controllers.
It's not related to attachments specifically, it's actually closely related to the portal mixin (but in reality, just the presence of an `access_token` field on the target model is enough).

Except now we also need it in `mail` (in this case for attachments), but it's not in the dependency for portal (and attachment doesn't have the portal mixin anyway).

I could move it into models.py override instead of http file?

Reply:
I put it originally in http file was because this method is only called in the context of a request for public user/with token before granting sudo access.

Reply:
Maybe it should be on models.py then, without model_name. Not sure why people try to put model methods as generic helpers. We have models, let us use them.